src/metabase/api/user.clj

```diff
@@ -299,17 +301,19 @@
     (api/checkp (not (db/exists? User, :%lower.email (if email (u/lower-case-en email) email), :id [:not= id]))
                 "email" (tru "Email address already associated to another user."))
     (db/transaction
-     ;; only user or self can update user info
+     ;; only superuser or self can update user info
      ;; implicitly prevent group manager from updating users' info
      (when (or (= id api/*current-user-id*)
                api/*is-superuser?*)
        (api/check-500
         (db/update! User id (u/select-keys-when body
-                                                :present (into #{:locale} (when api/*is-superuser?* [:login_attributes]))
-                                                :non-nil (set (concat [:first_name :last_name :email]
+                                                :present (into #{:first_name :last_name :locale} (when api/*is-superuser?* [:login_attributes]))
+                                                :non-nil (set (concat [:email]
```

Comment:
We only require `:email` to be non-nil now, so we can select the name keys when they're present. Nil names should be accepted now.

Reply:
I think the idiomatic way for constructing these sets would be
```
(cond-> #{:first_name :last_name :locale}
  api/*is-superuser?* (conj :login_attributes))
```
but at least we should not use two different ways for the construction. (I know it's not you who introduced them.)